### PR TITLE
added Xiaomi CGG1 sensor

### PR DESCRIPTION
**Pull request by Alex9779**
## Description:

**Related issue (if applicable):** none

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#357

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/xiaomi_cgg1/sensor.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from esphome.components import sensor, esp32_ble_tracker
+from esphome.const import CONF_BATTERY_LEVEL, CONF_HUMIDITY, CONF_MAC_ADDRESS, CONF_TEMPERATURE, \
+    UNIT_CELSIUS, ICON_THERMOMETER, UNIT_PERCENT, ICON_WATER_PERCENT, ICON_BATTERY, CONF_ID
+
+DEPENDENCIES = ['esp32_ble_tracker']
+AUTO_LOAD = ['xiaomi_ble']
+
+xiaomi_cgg1_ns = cg.esphome_ns.namespace('xiaomi_cgg1')
+XiaomiCGG1 = xiaomi_cgg1_ns.class_('XiaomiCGG1', esp32_ble_tracker.ESPBTDeviceListener,
+                                     cg.Component)
```
Comment (OttoWinter) [inline esphome/components/xiaomi_cgg1/sensor.py]: See https://travis-ci.org/esphome/esphome/jobs/597549291#L387

**Review by OttoWinter — COMMENTED** [review]
Ok, looks good. Just a lint comment and a bit of work in the docs. Thanks 👍

**Review by OttoWinter — APPROVED** [review]

**Review by OttoWinter — COMMENTED** [review]
Oh I just noticed, is this for the same device as https://github.com/esphome/esphome/pull/735?

**Comment by Alex9779** [comment]
crap just saw someone else added the device by its product name in #735

**Comment by Alex9779** [comment]
ummmm yes...

**Comment by Alex9779** [comment]
I wanted to add it that way too but "ClearGrass" is the name of a series of devices so I decided to go the way the LYWSD02 was added...

**Comment by Alex9779** [comment]
no idea how to handle this, IMHO the strucutre is already mixed because "MiJia" is also not just the device but a series.
So working with that name in the xiaomi_ble component to detect the device and its capabilities is just fine but there exist several sensors for each of that series.

**Comment by sermayoral** [comment]
Hi everybody.

I chose the same name that it has in Openmqttgateway in #735, but both names are OK. Getting this device working is the important thing 😃

**Comment by Alex9779** [comment]
Yeah this is just a question of philosophy... I vote for model names but this does not comply with the very start of the xiaomi module so...

**Comment by Alex9779** [comment]
Ok I did a little research and the device calle "MiJia Flora" is model "HHCCJCY01", what ESPHOME has as MiJia is "LYWSDCGQ", the "MiJia" in the xiaomi_ble module covers both and also the "CGG1" which is what is called "ClearGrass Digital Bluetooth Thermometer Humidity", and then we have the "LYWSD02".
OpenMQTTGateway implemented "LYWSD02" too with the model name.
*shrug*
I personally would go the model way, just mention the marketing names so that one can find it, rename "mijia" and "miflora" in the xiaomi_ble module and the xiaomi_mijia sensor to the model name...

**Comment by Alex9779** [comment]
closing in favor for #755

**Pull request closed without merging**